FBReader/src/main/java/org/bookshare/net/BookshareHttpOauth2Client.java

```diff
@@ -91,6 +92,39 @@ public JSONArray getReadingLists(String accessToken) throws Exception {
         return readingLists;
     }
 
+    public JSONArray postTitleToReadingList(String accessToken, String readingListId, String titleId) throws Exception {
+
+        String url = String.format("https://%s/v2/lists/%s/titles?api_key=%s",HOST_NAME, readingListId, API_KEY);
+
+        HttpsURLConnection urlConnection = createHttpsUrlConnection(url, POST_REQUESTE_METHOD);
+        setAccessToken(accessToken, urlConnection);
+
+        LinkedHashMap<String, String> formParameters = new LinkedHashMap<>();
+        formParameters.put("bookshareId", titleId);
+        writeFormParameters(formParameters, urlConnection);
+
+        final String rawResponseWithReadingLists = requestData(urlConnection);
+
+        JSONObject readingListJson = new JSONObject(rawResponseWithReadingLists);
+        JSONArray readingListsJsonArray = readingListJson.optJSONArray(JSON_CODE_LISTS);
+        if (readingListsJsonArray == null)
```

Comment:
Methods that have a return of type array, should return empty arrays, instead of null.  This will avoid null tests before looping the array.